ckanext/saml2/templates/user/edit_user_form.html

```diff
@@ -0,0 +1,77 @@
+{% import 'macros/form.html' as form %}
+
+{% if c.allow_user_changes|lower == 'true' %}
+    {% if not c.is_allow_update %}
+        {% set attrs = {'readonly': 'readonly'} %}
+    {% else %}
+        {% set attrs = {} %}
+    {% endif %}
+{% else %}
+    {% set attrs = {'readonly': 'readonly'} %}
+{% endif %}
```

Comment:
Try to avoid nested if-blocks and use `asbool` helper(i belive, that it already available in CKAN>2.4)